src/spidevioctl.rs

```diff
@@ -64,19 +65,51 @@ fn from_nix_result<T>(res: ::nix::Result<T>) -> io::Result<T> {
 /// last transfer might write some register values.
 /// ```
 #[allow(non_camel_case_types)]
-#[derive(Debug)]
+#[derive(Debug, Default)]
 #[repr(C)]
-pub struct spi_ioc_transfer {
-    pub tx_buf: u64,
-    pub rx_buf: u64,
-    pub len: u32,
+pub struct spi_ioc_transfer<'a, 'b> {
+    tx_buf: u64,
+    rx_buf: u64,
+    len: u32,
 
     // optional overrides
     pub speed_hz: u32,
     pub delay_usecs: u16,
     pub bits_per_word: u8,
     pub cs_change: u8,
     pub pad: u32,
+
+    tx_buf_ref: PhantomData<&'a [u8]>,
+    rx_buf_ref: PhantomData<&'b mut [u8]>,
```

Comment:
Nice use of PhantomData here.  Was concerned about the the transfer outlasting the lifetime of the buffers at first, but this handles that nicely.